crates/cdk-lnd/src/lib.rs

```diff
@@ -441,10 +434,9 @@ impl MintPayment for Lnd {
             ..Default::default()
         };
 
-        let invoice = self
-            .client
-            .lock()
-            .await
+        let mut client = self.client.clone();
+
+        let invoice = client
             .lightning()
```

Comment:
Instead, we should consider `&mut self` rather than `&self` in the function definition of  the trait and thus can do: 
```suggestion
let invoice = self.client
.lightning()
```

and IMO, using a client clone in the operations is causing the IT's to fail because calling an operation on the 
 client clone would not reflect changes in the `self.client`.